packages/qwik/src/core/render/container.ts

```diff
@@ -79,6 +79,17 @@ export const createContainerState = (containerEl: Element) => {
   return containerState;
 };
 
+const envDataFromContainer = (containerEl: Element): Record<string, any> => {
```

Comment:
envData is not container attributes, we are mixing semantics, not sure i like this, we are telling developers that envData is server only (for now), but it comes from `envData`, this breaks that semantic

Reply:
in our React codebase, we have plugins that can extract server data during render (via `<Provider>` mosty), and then populate a global state object that gets serialized to the client, which gets read by the Providers on the client.
This has proven very useful, and I think this `envData` is similar to that.